h2/src/main/org/h2/command/Parser.java

```diff
@@ -1573,7 +1573,10 @@ private Prepared parseHelp() {
     private Prepared parseShow() {
         ArrayList<Value> paramValues = Utils.newSmallArrayList();
         StringBuilder buff = new StringBuilder("SELECT ");
-        if (readIf("CLIENT_ENCODING")) {
+        if (readIf("ALL")) {
```

Comment:
1. You need to check compatibility mode here.

2. `ALL` is a keyword, it should read by its number.

Move it to the end and use something like
```Java
if (database.getMode().getEnum() == ModeEnum.PostgreSQL && (readIf(ALL)) {
}
```